File: src/main/java/com/now/nowbot/entity/OsuNameToIdLite.java
package com.now.nowbot.entity;

import org.hibernate.annotations.Type;

import javax.persistence.*;

@Entity
@Table(name = "osu_name_id", indexes = {
        @Index(name = "osufindname", columnList = "name"),
})

public class OsuNameToIdLite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String name;
    private Long uid;
    @Column(name = "idx")
    private Integer index;

    public OsuNameToIdLite() {
    }

    public OsuNameToIdLite(Long uid, String name, Integer index) {
        this.name = name.toUpperCase();
        this.uid = uid;
        this.index = index;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name.toUpperCase();
    }

    public Long getUid() {
        return uid;
    }

    public void setUid(Long uid) {
        this.uid = uid;
    }
}

File: src/main/java/com/now/nowbot/entity/OsuNameToIDLite.kt
package com.now.nowbot.entity

import jakarta.persistence.*

@Entity @Table(name = "osu_name_id",
    indexes = [
        // Index(name = "osufindname", columnList = "name"),
        // CREATE INDEX idx_osu_name_id_trgm_name ON osu_name_id USING gin (name gin_trgm_ops);
        Index(name = "osu_name_user_id", columnList = "user_id")
    ])
class OsuNameToIDLite {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "name", length = 32)
    var name: String? = null

    @Column(name = "user_id")
    var userID: Long? = null

    @Column(name = "index")
    private var index: Int? = null

    @Suppress("UNUSED")
    constructor()

    constructor(userID: Long?, name: String, index: Int?) {
        this.name = name
        this.userID = userID
        this.index = index
    }
}
